lib/prometheus/business_collector.rb

```diff
@@ -56,6 +56,12 @@ def initialize
     @client_systems_by_os = PrometheusExporter::Metric::Gauge.new(
       'client_systems_by_os', 'Systems by OS version (excluded Red Hat)'
     )
+    @total_reports_per_month = PrometheusExporter::Metric::Counter.new(
+      'total_reports_per_month', 'Reports uploaded per month'
```

Comment:
Technically this returns counts from current month. To get the right number, this number would have to be collected and **stored** exactly at the end of the month.

@akofink ideas how we can report this correctly? Considering that reports can be deleted, there would have to be some sort of incremental counter.

Example situation:
1. we got 10 reports at this moment; observing 10
2. one report will come in; observing 11
3. two reports are removed by user; observing 9
4. one report will come in; observing 10
```

There were 12 reports, however the last number observing number (to be collected) is 10.

Are there some means we can capture this easily?
Would a moving 1 month window be a "better" choice? (Reports in last 30 days)

Reply:
Well, the issue specifies the number of uploaded results by month, so a counter still might be a good solution, but I agree that the deletion messes this up :shrug:

Reply:
Reports don't even need to be removed by the user - they are removed if a new result replaces it for the same host and policy. I'd say, don't stress too much about the per month/year aspect of this metric. Let's push the data into prometheus as reports per time and use grafana to see trends over larger amounts of time (we have a card to ask platform to store > 30 days worth of metrics in prometheus). We already have sidekiq job metrics which counts how many ParseReportJobs per time frame come into Compliance (`compliance_sidekiq_jobs_total{job_name="ParseReportJobs"}`). I don't see why we couldn't use that for this metric. It's a counter, as you intuited would be best for this metric. :+1: